### ISSUE TITLE
Style Question: Why module prefixes over imports?

### ISSUE THREAD

**Issue by willemneal** [stellar/soroban-cli#243]
For example, throughout the repo, `jsonrpc` is prefixed in favor of importing types from the module. Any reason for this?

**Comment by leighmcculloch** [comment]
I think developers can make a choice about what level of import to use such that code communicates sufficient intent. It really depend on the context, and none of this is what I would define as mandatory or required to our style, we're just trying to present the code in a form that lends itself to increased readability. I think we're inclined to elevate code clarity as valuable over any sort of verbosity vs brevity measure. Of course verbosity and brevity impact code clarity too.

Using `jsonrpc`, which you mentioned, as an example:

The `jsonrpc` module contains a few types with reasonably generic names, such as `Id`. It can be helpful at call sites to include that context, so that a reader doesn't only read `Id`, but reads `jsonrpc::Id` and therefore has that additional context.

Our use of the `serde_json` crate is also a good example:

The `serde_json` package contains a function `to_string` and we call it in a number of places. We could import `to_string` and call it, but a loan `to_string` function does very little to communicate sufficient intent and context, and so we often fully qualify it as `serde_json::to_string`.

Some community discussion on the matter:
- https://internals.rust-lang.org/t/does-the-avoid-prefixing-the-type-with-the-name-of-the-module-style-rule-still-apply/12819

**Comment by willemneal** [comment]
Yeah I understand for generic names like `serde_json` but for internal types it feels overly verbose and with tools like RA finding definitions is a easy.

**Comment by leighmcculloch** [comment]
Agreed it is pretty verbose. I don't feel strongly about how the `jsonrpc` module is used here as qualified or used.

cc @paulbellamy in case you disagree.
